tls/s2n_handshake_transcript.c

```diff
@@ -52,7 +52,9 @@ int s2n_conn_post_handshake_hashes_update(struct s2n_connection *conn)
         /* If we are sending a retry request, we didn't decide on a key share. There are no secrets to handle. */
         break;
     case SERVER_HELLO:
-        GUARD(s2n_tls13_handle_handshake_secrets(conn));
+        if (s2n_hello_retry_validate(conn) != S2N_SUCCESS) {
```

Comment:
Wouldn't a call to s2n_is_hello_retry_message make more sense? We're not re-validating the message, we're checking that we're on the right message.

Reply:
At this stage ACTIVE_MESSAGE(conn)  is not `HELLO_RETRY_MSG` and it is `SERVER_HELLO`

Reply:
Then when would you ever hit this code with s2n_conn_get_current_message_type == SERVER_HELLO and s2n_hello_retry_validate() == S2N_SUCCESS?

Reply:
Removing this condition, it is not required.